src/Router/AppShell.tsx

```diff
@@ -4,7 +4,15 @@ import { ContextProvider } from "../Components/Artsy"
 import { AppShellProps } from "./types"
 
 export const AppShell: SFC<AppShellProps> = props => {
-  const { loadableState, data, children, provide } = props
+  const { loadableState, data = {}, children, provide } = props
+
+  let hydrationData
+  try {
+    hydrationData = serialize(data, { isJSON: true })
```

Comment:
👍

Reply:
this file is so ugly... really need to refactor all of this

Reply:
We'll get there.